### PR TITLE
microwave can be turned on with signal

### PR DESCRIPTION
**Pull request by deltanedas**
## About the PR
title

## Why / Balance
- lets you automate kitchen a little bit, e.g. with #20026 have a 15s timer control a clock that activates a 10s microwave to get an efficient breab pipeline, 5s window to put in new dough rinse and repeat
- let sussy/rev make remotely activated microwave bombs real

## Technical details
simple

## Media

hampter roulette + battery bomb

https://github.com/space-wizards/space-station-14/assets/39013340/b1c53475-3ada-4f8d-9739-2a07c3a2109a

- [X] I have added screenshots/videos to this PR showcasing its changes ingame, **or** this PR does not require an ingame showcase

## Breaking changes
no

**Changelog**
no cl no fun

### CODE REVIEW

**Comment by Eden077** [comment]
Man... Microwaving hamlet has never been easier...

**Comment by NancokPS2** [comment]
Time to play SAW with Hammy

File: Content.Server/Kitchen/EntitySystems/MicrowaveSystem.cs [hunk 1]
```diff
@@ -175,6 +183,7 @@ private void SubtractContents(MicrowaveComponent component, FoodRecipePrototype
         private void OnInit(EntityUid uid, MicrowaveComponent component, ComponentInit ags)
         {
             component.Storage = _container.EnsureContainer<Container>(uid, "microwave_entity_container");
+            _deviceLink.EnsureSinkPorts(uid, component.OnPort);
```
Comment (metalgearsloth) [inline Content.Server/Kitchen/EntitySystems/MicrowaveSystem.cs:186]: I think these should go on mapinit not componentinit.

Comment (deltanedas) [reply Content.Server/Kitchen/EntitySystems/MicrowaveSystem.cs:186]: will change soon

Comment (deltanedas) [reply Content.Server/Kitchen/EntitySystems/MicrowaveSystem.cs]: the storage actually had to be in init, it breaks game and tests

**Comment by EmoGarbage404** [comment]
i don't think this makes much sense and remotely detonating bombs in microwaves is kinda a dumb powergamey feature. Especially since the microwaves aren't really "high tech" (they look like they were stolen out the back of a church kitchen), it doesn't make sense that you could do any kind of advanced linking with them.

**Comment by deltanedas** [comment]
in the future it should require gluing a signal trigger to any random device yes

**Comment by notquitehadouken** [comment]
> i don't think this makes much sense and remotely detonating bombs in microwaves is kinda a dumb powergamey feature. Especially since the microwaves aren't really "high tech" (they look like they were stolen out the back of a church kitchen), it doesn't make sense that you could do any kind of advanced linking with them.

i want to make people terrified of microwaves

**Pull request merged**